### PR TITLE
Integrating A* search

### PR DESCRIPTION
**Pull request by agarwalishika**

### CODE REVIEW

File: tests/test_astar.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import sys
+sys.path.append('../')
```
Comment (ZikangXiong) [inline tests/test_astar.py]: If you are using conda, you can `conda develop /path/to/two_d_nav_gym`. If you are using pip, you can add a `setup.py` to `/path/to/two_d_nav_gym`, and use `pip install -e /path/to/two_d_nav_gym`. Then this path will be appended to the python path. 

sys.path.append is not the common practice.

File: tests/test_astar.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+def create_board(lines, obs_list):
+  board_size = c.BOARD_SIZE
+  board = np.zeros(shape = (board_size, board_size))
+  board = define_walls(board, lines)
+  board = define_obstables(board, obs_list)
+  return board, board_size, board_size
+
+def deconstructPath(path, row, col):
```
Comment (ZikangXiong) [inline tests/test_astar.py]: code style issue, the function name should be snake case.

File: tests/test_astar.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    elif current.col+c.STEP == next.col:
+      actions.append("down")
+    elif current.row-c.STEP == next.row:
+      actions.append("left")
+    elif current.row+c.STEP == next.row:
+      actions.append("right")
+    current = next
+  return actions
+
+def runSearch(board, board_rows, board_cols):
```
Comment (ZikangXiong) [inline tests/test_astar.py]: code style

File: tests/test_astar.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  goal = Location(c.GOAL_POS_X, c.GOAL_POS_Y)
+
+  state = State(board.flatten(), start, board_rows, board_cols)
+  ag = Agent(state)
+  path, pathCost = ag.a_star_search(start, goal)
+
+  actions = deconstructPath(path, board_rows, board_cols)
+  return actions
+
+def printActions(actions):
```
Comment (ZikangXiong) [inline tests/test_astar.py]: code style

File: tests/test_astar.py (hunk 1 above)
Comment (agarwalishika) [reply tests/test_astar.py]: resolved

File: tests/test_astar.py (hunk 2 above)
Comment (agarwalishika) [reply tests/test_astar.py]: resolved

File: tests/test_astar.py (hunk 3 above)
Comment (agarwalishika) [reply tests/test_astar.py]: resolved

File: tests/test_astar.py (hunk 4 above)
Comment (agarwalishika) [reply tests/test_astar.py]: resolved

**Pull request merged**